library/Vanilla/Embeds/GiphyEmbed.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0
+ */
+
+namespace Vanilla\Embeds;
+
+use Exception;
+
+/**
+ * giphy Embed.
+ */
+class GiphyEmbed extends Embed {
+
+    /** @inheritdoc */
+    protected $domains = ['giphy.com', 'media.giphy.com', 'gph.is'];
+
+    /**
+     * giphyEmbed constructor.
+     */
+    public function __construct() {
+        parent::__construct('giphy', 'gif');
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function matchUrl(string $url) {
+        $data = [];
+        $domain = parse_url($url, PHP_URL_HOST);
+
+        if ($this->isNetworkEnabled()) {
+
+            $post = $this->parseURL($url);
+
+            $oembedData = $this->oembed("https://giphy.com/services/oembed?url=".urlencode($url));
+            if ($oembedData) {
+                $data = $this->normalizeOembed($oembedData);
+            }
+
+            // Obtain the real post ID from media url provided by oembed call
+            if ($domain == 'gph.is') {
+                $post = $this->parseURL($data['url']);
```

Comment:
With this logic, you're assuming `$data` is populated. However, if `$oembedData` were falsy, `$data` would be empty.